### PR TITLE
[Partitioner] Enumerate partitions by iterating partition ids

### PR DESCRIPTION
**Pull request by ghost**
Currently, we get all partition id by iterating assignment whose size is same as the number of nodes in graph. But we can reach same results by iterating partitions_by_id whose size is much smaller than the nodes number. Assume the number of nodes is N, the number of partitions is P, the time complexity decrease from O(N * N) to O(N * P) after this patch.

### CODE REVIEW

File: torch/fx/passes/infra/partitioner.py
Comment (Skylion007) [inline torch/fx/passes/infra/partitioner.py:209]:
```suggestion
            merge_candidates.update(dict.fromkeys(partitions_by_id))
```
may be marginally faster

Comment (lingzhi98) [reply torch/fx/passes/infra/partitioner.py:209]: I tested these two versions:
```
import time
cur = time.time()
for i in range(10000):
    a = {}
    b = {}
    for j in range(10):
        a[i + j] = j
    # b.update(dict.fromkeys(a))
    for key in a:
        b[key] = None
print(time.time() - cur)
```
The result show that for-loop is a little faster than dict.fromkeys. Maybe dict.fromkeys is faster than for-loop if the size of ```a``` is large enough? But I think we can assume the number of partitions is small.

**Comment by lingzhi98** [comment]
@tarun292 can you take a look? Thanks!

**Comment by carsonwang** [comment]
Adding some background info here. In one of our use cases, we have observed pass_propose_partitions took more than 1 minute. We have proposed totally 4 PRs(#136608, #136616, #136614, #136598) to optimize it and they achieved up to 11x speedup in our test. @tarun292 @SherlockNoMad appreciate if you can take a look if these changes make sense. This will be very useful to unblock our use case.

File: torch/fx/passes/infra/partitioner.py
Comment (tarun292) [reply torch/fx/passes/infra/partitioner.py:217]: @lingzhi98 we have various internal use cases where the number of partitions generated are >100 in some cases. If `b.update(dict.fromkeys(a)` is faster for > 10 elements do you mind doing this change? Other than that this PR looks good.

Comment (lingzhi98) [reply torch/fx/passes/infra/partitioner.py]: Done.

**Review by tarun292 — DISMISSED** [review]

**Comment by xwu99** [comment]
@ezyang Could you help to review and merge this first as it's a one line change and easy to reason.?

**Comment by huydhn** [comment]
@pytorchbot revert -m 'Sorry for reverting your change but it fails an executorch test https://github.com/pytorch/executorch/blob/main/exir/backend/test/test_graph_partition.py#L114-L175' -c ghfirst

Here is the failure:

```
======================================================================
FAIL: test_partition_list_without_op_support_two_partitions (executorch.exir.backend.test.test_graph_partition.TestGraphPartition)
check graph will be divided into 2 parts when the supported modules is provided, but OpeartorSupportBase is not provideds
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/data/sandcastle/boxes/eden-trunk-hg-full-fbsource/buck-out/v2/gen/fbcode/4ac7a641bc650342/executorch/exir/backend/test/__test_graph_partition__/test_graph_partition#link-tree/executorch/exir/backend/test/test_graph_partition.py", line 174, in test_partition_list_without_op_support_two_partitions
    self.assertEqual(node_list_1, partition_1)
AssertionError: Lists differ: ['ate[47 chars]ault_3', 'aten_relu_default', 'p_conv3_bias'] != ['ate[47 chars]ault_3', 'aten_relu_default', 'p_conv3_bias', 'p_conv3_weight']

Second list contains 1 additional elements.
First extra element 4:
'p_conv3_weight'

  ['aten_convolution_default_2',
   'aten_convolution_default_3',
   'aten_relu_default',
-  'p_conv3_bias']
?                ^

+  'p_conv3_bias',
?                ^

+  'p_conv3_weight']

----------------------------------------------------------------------
Ran 1 test in 5.707s
```

Maybe the test needs to be updated.  If it does, please submit a PR to ExecuTorch and I could help facilitate its review

**Comment by ezyang** [comment]
yes sgtm @huydhn can you help?

**Comment by lingzhi98** [comment]
Yes, they are different as I mentioned. For this failed test, the conv3_weight will be merged into partition_2 due to different order. But we can not make sure the merge_candidates order is same as previous. Do you have any idea about it or any concern about performance? @tarun292

**Comment by huydhn** [comment]
Let me try to import this first.  On the other hand, I think more ExecuTorch failures shows up.  One of them looks relevant

```
======================================================================
FAIL: test_f16 (executorch.backends.xnnpack.test.models.llama2_et_example.TestLlama2ETExample)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/data/sandcastle/boxes/eden-trunk-hg-full-fbsource/buck-out/v2/gen/fbcode/4ac7a641bc650342/executorch/backends/xnnpack/test/__test_xnnpack_models__/test_xnnpack_models#link-tree/executorch/backends/xnnpack/test/models/llama2_et_example.py", line 20, in test_f16
    self._test(torch.float16)
  File "/data/sandcastle/boxes/eden-trunk-hg-full-fbsource/buck-out/v2/gen/fbcode/4ac7a641bc650342/executorch/backends/xnnpack/test/__test_xnnpack_models__/test_xnnpack_models#link-tree/executorch/backends/xnnpack/test/models/llama2_et_example.py", line 42, in _test
    .to_edge_transform_and_lower()
  File "/data/sandcastle/boxes/eden-trunk-hg-full-fbsource/buck-out/v2/gen/fbcode/4ac7a641bc650342/executorch/backends/xnnpack/test/__test_xnnpack_models__/test_xnnpack_models#link-tree/executorch/backends/xnnpack/test/tester/tester.py", line 570, in to_edge_transform_and_lower
    return self._run_stage(to_edge_and_transform_stage or ToEdgeTransformAndLower())
  File "/data/sandcastle/boxes/eden-trunk-hg-full-fbsource/buck-out/v2/gen/fbcode/4ac7a641bc650342/executorch/backends/xnnpack/test/__test_xnnpack_models__/test_xnnpack_models#link-tree/executorch/backends/xnnpack/test/tester/tester.py", line 545, in _run_stage
    stage_instance.run(prev_stage_artifact, inputs=inputs)
  File "/data/sandcastle/boxes/eden-trunk-hg-full-fbsource/buck-out/v2/gen/fbcode/4ac7a641bc650342/executorch/backends/xnnpack/test/__test_xnnpack_models__/test_xnnpack_models#link-tree/executorch/backends/xnnpack/test/tester/tester.py", line 301, in run
    self.edge_dialect_program = to_edge_transform_and_lower(
  File "/data/sandcastle/boxes/eden-trunk-hg-full-fbsource/buck-out/v2/gen/fbcode/4ac7a641bc650342/executorch/backends/xnnpack/test/__test_xnnpack_models__/test_xnnpack_models#link-tree/executorch/exir/program/_program.py", line 1035, in to_edge_transform_and_lower
    edge_manager = edge_manager.to_backend({name: curr_partitioner})
  File "/data/sandcastle/boxes/eden-trunk-hg-full-fbsource/buck-out/v2/gen/fbcode/4ac7a641bc650342/executorch/backends/xnnpack/test/__test_xnnpack_models__/test_xnnpack_models#link-tree/executorch/exir/program/_program.py", line 1286, in to_backend
    new_edge_programs[name] = to_backend(program, partitioner[name])
  File "/usr/local/fbcode/platform010/lib/python3.10/functools.py", line 889, in wrapper
    return dispatch(args[0].__class__)(*args, **kw)
  File "/data/sandcastle/boxes/eden-trunk-hg-full-fbsource/buck-out/v2/gen/fbcode/4ac7a641bc650342/executorch/backends/xnnpack/test/__test_xnnpack_models__/test_xnnpack_models#link-tree/executorch/exir/backend/backend_api.py", line 396, in _
    tagged_graph_module = _partition_and_lower(
  File "/data/sandcastle/boxes/eden-trunk-hg-full-fbsource/buck-out/v2/gen/fbcode/4ac7a641bc650342/executorch/backends/xnnpack/test/__test_xnnpack_models__/test_xnnpack_models#link-tree/executorch/exir/backend/backend_api.py", line 319, in _partition_and_lower
    partitioned_module = _partition_and_lower_one_graph_module(
  File "/data/sandcastle/boxes/eden-trunk-hg-full-fbsource/buck-out/v2/gen/fbcode/4ac7a641bc650342/executorch/backends/xnnpack/test/__test_xnnpack_models__/test_xnnpack_models#link-tree/executorch/exir/backend/backend_api.py", line 222, in _partition_and_lower_one_graph_module
    submodule, call_module_node = create_submodule_from_nodes(
  File "/data/sandcastle/boxes/eden-trunk-hg-full-fbsource/buck-out/v2/gen/fbcode/4ac7a641bc650342/executorch/backends/xnnpack/test/__test_xnnpack_models__/test_xnnpack_models#link-tree/executorch/exir/lowered_backend_module.py", line 744, in create_submodule_from_nodes
    sub_gm, orig_inputs, orig_outputs = fuse_as_graphmodule(
  File "/data/sandcastle/boxes/eden-trunk-hg-full-fbsource/buck-out/v2/gen/fbcode/4ac7a641bc650342/executorch/backends/xnnpack/test/__test_xnnpack_models__/test_xnnpack_models#link-tree/torch/fx/passes/utils/fuser_utils.py", line 139, in fuse_as_graphmodule
    assert validate_partition(nodes), "Invalid partition, found dependency cycles"
AssertionError: Invalid partition, found dependency cycles

----------------------------------------------------------------------
Ran 1 test in 89.214s
```

**Comment by lingzhi98** [comment]
> Let me try to import this first. On the other hand, I think more ExecuTorch failures shows up. One of them looks relevant
> 
> ```
> ======================================================================
> FAIL: test_f16 (executorch.backends.xnnpack.test.models.llama2_et_example.TestLlama2ETExample)
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/data/sandcastle/boxes/eden-trunk-hg-full-fbsource/buck-out/v2/gen/fbcode/4ac7a641bc650342/executorch/backends/xnnpack/test/__test_xnnpack_models__/test_xnnpack_models#link-tree/executorch/backends/xnnpack/test/models/llama2_et_example.py", line 20, in test_f16
>     self._test(torch.float16)
>   File "/data/sandcastle/boxes/eden-trunk-hg-full-fbsource/buck-out/v2/gen/fbcode/4ac7a641bc650342/executorch/backends/xnnpack/test/__test_xnnpack_models__/test_xnnpack_models#link-tree/executorch/backends/xnnpack/test/models/llama2_et_example.py", line 42, in _test
>     .to_edge_transform_and_lower()
>   File "/data/sandcastle/boxes/eden-trunk-hg-full-fbsource/buck-out/v2/gen/fbcode/4ac7a641bc650342/executorch/backends/xnnpack/test/__test_xnnpack_models__/test_xnnpack_models#link-tree/executorch/backends/xnnpack/test/tester/tester.py", line 570, in to_edge_transform_and_lower
>     return self._run_stage(to_edge_and_transform_stage or ToEdgeTransformAndLower())
>   File "/data/sandcastle/boxes/eden-trunk-hg-full-fbsource/buck-out/v2/gen/fbcode/4ac7a641bc650342/executorch/backends/xnnpack/test/__test_xnnpack_models__/test_xnnpack_models#link-tree/executorch/backends/xnnpack/test/tester/tester.py", line 545, in _run_stage
>     stage_instance.run(prev_stage_artifact, inputs=inputs)
>   File "/data/sandcastle/boxes/eden-trunk-hg-full-fbsource/buck-out/v2/gen/fbcode/4ac7a641bc650342/executorch/backends/xnnpack/test/__test_xnnpack_models__/test_xnnpack_models#link-tree/executorch/backends/xnnpack/test/tester/tester.py", line 301, in run
>     self.edge_dialect_program = to_edge_transform_and_lower(
>   File "/data/sandcastle/boxes/eden-trunk-hg-full-fbsource/buck-out/v2/gen/fbcode/4ac7a641bc650342/executorch/backends/xnnpack/test/__test_xnnpack_models__/test_xnnpack_models#link-tree/executorch/exir/program/_program.py", line 1035, in to_edge_transform_and_lower
>     edge_manager = edge_manager.to_backend({name: curr_partitioner})
>   File "/data/sandcastle/boxes/eden-trunk-hg-full-fbsource/buck-out/v2/gen/fbcode/4ac7a641bc650342/executorch/backends/xnnpack/test/__test_xnnpack_models__/test_xnnpack_models#link-tree/executorch/exir/program/_program.py", line 1286, in to_backend
>     new_edge_programs[name] = to_backend(program, partitioner[name])
>   File "/usr/local/fbcode/platform010/lib/python3.10/functools.py", line 889, in wrapper
>     return dispatch(args[0].__class__)(*args, **kw)
>   File "/data/sandcastle/boxes/eden-trunk-hg-full-fbsource/buck-out/v2/gen/fbcode/4ac7a641bc650342/executorch/backends/xnnpack/test/__test_xnnpack_models__/test_xnnpack_models#link-tree/executorch/exir/backend/backend_api.py", line 396, in _
>     tagged_graph_module = _partition_and_lower(
>   File "/data/sandcastle/boxes/eden-trunk-hg-full-fbsource/buck-out/v2/gen/fbcode/4ac7a641bc650342/executorch/backends/xnnpack/test/__test_xnnpack_models__/test_xnnpack_models#link-tree/executorch/exir/backend/backend_api.py", line 319, in _partition_and_lower
>     partitioned_module = _partition_and_lower_one_graph_module(
>   File "/data/sandcastle/boxes/eden-trunk-hg-full-fbsource/buck-out/v2/gen/fbcode/4ac7a641bc650342/executorch/backends/xnnpack/test/__test_xnnpack_models__/test_xnnpack_models#link-tree/executorch/exir/backend/backend_api.py", line 222, in _partition_and_lower_one_graph_module
>     submodule, call_module_node = create_submodule_from_nodes(
>   File "/data/sandcastle/boxes/eden-trunk-hg-full-fbsource/buck-out/v2/gen/fbcode/4ac7a641bc650342/executorch/backends/xnnpack/test/__test_xnnpack_models__/test_xnnpack_models#link-tree/executorch/exir/lowered_backend_module.py", line 744, in create_submodule_from_nodes
>     sub_gm, orig_inputs, orig_outputs = fuse_as_graphmodule(
>   File "/data/sandcastle/boxes/eden-trunk-hg-full-fbsource/buck-out/v2/gen/fbcode/4ac7a641bc650342/executorch/backends/xnnpack/test/__test_xnnpack_models__/test_xnnpack_models#link-tree/torch/fx/passes/utils/fuser_utils.py", line 139, in fuse_as_graphmodule
>     assert validate_partition(nodes), "Invalid partition, found dependency cycles"
> AssertionError: Invalid partition, found dependency cycles
> 
> ----------------------------------------------------------------------
> Ran 1 test in 89.214s
> ```

This failure is caused by exist bug in partitioner. We can see these 2 parts: https://github.com/pytorch/pytorch/blob/main/torch/fx/passes/infra/partitioner.py#L123 and https://github.com/pytorch/pytorch/blob/main/torch/fx/passes/utils/fuser_utils.py#L63. The former doesn't consider unsupported node when find dependency cycle but the latter does. There exists 2 ways to fix this: stricter check for the former (the latest patch choose this, but cause more partitions) or looser check of the latter. Maybe we should more discussion with this. @ezyang @tarun292 @huydhn

**Review by ezyang — DISMISSED** [review]

**Comment by lingzhi98** [comment]
It seems some workflow requires approval. Can anyone help?

**Comment by xwu99** [comment]
> > Let me try to import this first. On the other hand, I think more ExecuTorch failures shows up. One of them looks relevant
> > ```
> > ======================================================================
> > FAIL: test_f16 (executorch.backends.xnnpack.test.models.llama2_et_example.TestLlama2ETExample)
> > ----------------------------------------------------------------------
> > Traceback (most recent call last):
> >   File "/data/sandcastle/boxes/eden-trunk-hg-full-fbsource/buck-out/v2/gen/fbcode/4ac7a641bc650342/executorch/backends/xnnpack/test/__test_xnnpack_models__/test_xnnpack_models#link-tree/executorch/backends/xnnpack/test/models/llama2_et_example.py", line 20, in test_f16
> >     self._test(torch.float16)
> >   File "/data/sandcastle/boxes/eden-trunk-hg-full-fbsource/buck-out/v2/gen/fbcode/4ac7a641bc650342/executorch/backends/xnnpack/test/__test_xnnpack_models__/test_xnnpack_models#link-tree/executorch/backends/xnnpack/test/models/llama2_et_example.py", line 42, in _test
> >     .to_edge_transform_and_lower()
> >   File "/data/sandcastle/boxes/eden-trunk-hg-full-fbsource/buck-out/v2/gen/fbcode/4ac7a641bc650342/executorch/backends/xnnpack/test/__test_xnnpack_models__/test_xnnpack_models#link-tree/executorch/backends/xnnpack/test/tester/tester.py", line 570, in to_edge_transform_and_lower
> >     return self._run_stage(to_edge_and_transform_stage or ToEdgeTransformAndLower())
> >   File "/data/sandcastle/boxes/eden-trunk-hg-full-fbsource/buck-out/v2/gen/fbcode/4ac7a641bc650342/executorch/backends/xnnpack/test/__test_xnnpack_models__/test_xnnpack_models#link-tree/executorch/backends/xnnpack/test/tester/tester.py", line 545, in _run_stage
> >     stage_instance.run(prev_stage_artifact, inputs=inputs)
> >   File "/data/sandcastle/boxes/eden-trunk-hg-full-fbsource/buck-out/v2/gen/fbcode/4ac7a641bc650342/executorch/backends/xnnpack/test/__test_xnnpack_models__/test_xnnpack_models#link-tree/executorch/backends/xnnpack/test/tester/tester.py", line 301, in run
> >     self.edge_dialect_program = to_edge_transform_and_lower(
> >   File "/data/sandcastle/boxes/eden-trunk-hg-full-fbsource/buck-out/v2/gen/fbcode/4ac7a641bc650342/executorch/backends/xnnpack/test/__test_xnnpack_models__/test_xnnpack_models#link-tree/executorch/exir/program/_program.py", line 1035, in to_edge_transform_and_lower
> >     edge_manager = edge_manager.to_backend({name: curr_partitioner})
> >   File "/data/sandcastle/boxes/eden-trunk-hg-full-fbsource/buck-out/v2/gen/fbcode/4ac7a641bc650342/executorch/backends/xnnpack/test/__test_xnnpack_models__/test_xnnpack_models#link-tree/executorch/exir/program/_program.py", line 1286, in to_backend
> >     new_edge_programs[name] = to_backend(program, partitioner[name])
> >   File "/usr/local/fbcode/platform010/lib/python3.10/functools.py", line 889, in wrapper
> >     return dispatch(args[0].__class__)(*args, **kw)
> >   File "/data/sandcastle/boxes/eden-trunk-hg-full-fbsource/buck-out/v2/gen/fbcode/4ac7a641bc650342/executorch/backends/xnnpack/test/__test_xnnpack_models__/test_xnnpack_models#link-tree/executorch/exir/backend/backend_api.py", line 396, in _
> >     tagged_graph_module = _partition_and_lower(
> >   File "/data/sandcastle/boxes/eden-trunk-hg-full-fbsource/buck-out/v2/gen/fbcode/4ac7a641bc650342/executorch/backends/xnnpack/test/__test_xnnpack_models__/test_xnnpack_models#link-tree/executorch/exir/backend/backend_api.py", line 319, in _partition_and_lower
> >     partitioned_module = _partition_and_lower_one_graph_module(
> >   File "/data/sandcastle/boxes/eden-trunk-hg-full-fbsource/buck-out/v2/gen/fbcode/4ac7a641bc650342/executorch/backends/xnnpack/test/__test_xnnpack_models__/test_xnnpack_models#link-tree/executorch/exir/backend/backend_api.py", line 222, in _partition_and_lower_one_graph_module
> >     submodule, call_module_node = create_submodule_from_nodes(
> >   File "/data/sandcastle/boxes/eden-trunk-hg-full-fbsource/buck-out/v2/gen/fbcode/4ac7a641bc650342/executorch/backends/xnnpack/test/__test_xnnpack_models__/test_xnnpack_models#link-tree/executorch/exir/lowered_backend_module.py", line 744, in create_submodule_from_nodes
> >     sub_gm, orig_inputs, orig_outputs = fuse_as_graphmodule(
> >   File "/data/sandcastle/boxes/eden-trunk-hg-full-fbsource/buck-out/v2/gen/fbcode/4ac7a641bc650342/executorch/backends/xnnpack/test/__test_xnnpack_models__/test_xnnpack_models#link-tree/torch/fx/passes/utils/fuser_utils.py", line 139, in fuse_as_graphmodule
> >     assert validate_partition(nodes), "Invalid partition, found dependency cycles"
> > AssertionError: Invalid partition, found dependency cycles
> > 
> > ----------------------------------------------------------------------
> > Ran 1 test in 89.214s
> > ```
> 
> This failure is caused by exist bug in partitioner. We can see these 2 parts: https://github.com/pytorch/pytorch/blob/main/torch/fx/passes/infra/partitioner.py#L123 and https://github.com/pytorch/pytorch/blob/main/torch/fx/passes/utils/fuser_utils.py#L63. The former doesn't consider unsupported node when find dependency cycle but the latter does. There exists 2 ways to fix this: stricter check for the former (the latest patch choose this, but cause more partitions) or looser check of the latter. Maybe we need more discussion with this. @ezyang @tarun292 @huydhn

@lingzhi98 could you elaborate on this? What are the bugs you are refering to, is it realted to this PR?

**Comment by kit1980** [comment]
One on the errors I see is `RuntimeError: Weight and Input should both be quantized`
Someone from ExecuTorch should help with re-landing this.

**Comment by huydhn** [comment]
@ptyorchbot rebase -b main

**Comment by mcr229** [comment]
Hi I took a look at this, and there were some internal failures because constant data nodes were not merged with the partitions of the user. For example, Consider the following graph (Note these nodes are just in the executorch namespace which is why they are named differently):

```
def forward(self, p_conv1_weight, p_conv1_bias, p_conv2_weight, p_conv2_bias, p_conv3_weight, p_conv3_bias, x):
    aten_convolution_default = executorch_exir_dialects_edge__ops_aten_convolution_default(x, p_conv1_weight, p_conv1_bias, [1, 1], [0, 0], [1, 1], False, [0, 0], 1);  x = p_conv1_weight = p_conv1_bias = None
    aten_convolution_default_1 = executorch_exir_dialects_edge__ops_aten_convolution_default(aten_convolution_default, p_conv2_weight, p_conv2_bias, [1, 1], [0, 0], [1, 1], False, [0, 0], 1);  p_conv2_weight = p_conv2_bias = None
    aten_add_tensor = executorch_exir_dialects_edge__ops_aten_add_Tensor(aten_convolution_default, aten_convolution_default_1);  aten_convolution_default = aten_convolution_default_1 = None
    aten_convolution_default_2 = executorch_exir_dialects_edge__ops_aten_convolution_default(aten_add_tensor, p_conv3_weight, p_conv3_bias, [1, 1], [0, 0], [1, 1], False, [0, 0], 1);  aten_add_tensor = None
    aten_convolution_default_3 = executorch_exir_dialects_edge__ops_aten_convolution_default(aten_convolution_default_2, p_conv3_weight, p_conv3_bias, [1, 1], [0, 0], [1, 1], False, [0, 0], 1);  p_conv3_weight = p_conv3_bias = None
    aten_max_dim = executorch_exir_dialects_edge__ops_aten_max_dim(aten_convolution_default_2, 2);  aten_convolution_default_2 = None
    getitem = aten_max_dim[0];  aten_max_dim = None
    aten_max_dim_1 = executorch_exir_dialects_edge__ops_aten_max_dim(aten_convolution_default_3, 2);  aten_convolution_default_3 = None
    getitem_1 = aten_max_dim_1[0];  aten_max_dim_1 = None
    aten_sub_tensor = executorch_exir_dialects_edge__ops_aten_sub_Tensor(getitem_1, getitem);  getitem_1 = getitem = None
    aten_relu_default = executorch_exir_dialects_edge__ops_aten_relu_default(aten_sub_tensor);  aten_sub_tensor = None
    return (aten_relu_default,)
```

After your change we see the following partitions:
```
{aten_relu_default: None}, {aten_convolution_default: None, aten_convolution_default_1: None, aten_add_tensor: None, aten_convolution_default_2: None, aten_convolution_default_3: None}, {p_conv1_weight: None, p_conv1_bias: None, p_conv2_weight: None, p_conv2_bias: None, p_conv3_weight: None, p_conv3_bias: None}]
```

note that the weights and bias's of the convs are in separate partitions. Before we saw the following partitions:

```
{aten_relu_default: None}, {p_conv1_weight: None, p_conv1_bias: None, p_conv2_weight: None, p_conv2_bias: None, p_conv3_weight: None, p_conv3_bias: None, aten_convolution_default: None, aten_convolution_default_1: None, aten_add_tensor: None, aten_convolution_default_2: None, aten_convolution_default_3: None}]
```

Let me debug a little more to see why this happens

**Comment by mcr229** [comment]
I tested the new commits since the revert internally and seemed to pass those failing tests for executorch

**Comment by lingzhi98** [comment]
> I tested the new commits since the revert internally and seemed to pass those failing tests for executorch

These failure is caused by different merged candidates order. From python 3.6, the order of dict.keys() order is same as insertion order. So we can keep same order as previous even though we don't iterate `assignment` dict.

**Review by mcr229 — APPROVED** [review]
Tested internally and looks clean now

**Comment by lingzhi98** [comment]
Can anyone help to reland this PR? Thanks.

**Pull request closed without merging**
